journeyapps/powersync-supabase-flutter-todolist-demo
# [Feature] Attachments example

## Description
Create a demo showcasing how adding attachments could work, the attachment queue logic will be extracted and put into a lib once we are happy with it which is why this is just a draft PR for now to get feedback before adding library complexity.

## Work Done
* Added what would go into the library in `attachments_queue` folder (to be extracted)
* Added implementation of `attachments_queue` and widgets used in `attachments` folder so everything is in one place
* Added take photo and showing photo functionality to todo items

## How to test
* While having no storage bucket set make sure no photos or photo taking functionality is visible when in a todo list
* Add a storage bucket to AppConfig (easiest way to set this up is in supabase and use the name of the storage)
* Now when in a todo list you should be able to add a photo. This photo should be uploaded to storage - confirm this.
* Using another device log into the app and confirm that the picture is downloaded to that device.
* Go offline and add and remove photos and go back online - confirm everything uploads and deletes as expected.

## Screenshots
**No Storage Bucket Set**
<img width="445" alt="image" src="https://github.com/powersync-ja/powersync-supabase-flutter-todolist-demo/assets/46312751/82d7d2e1-e6ad-4bee-8e50-779db30d884f">

**Storage bucket set with photos** 
<img width="426" alt="image" src="https://github.com/powersync-ja/powersync-supabase-flutter-todolist-demo/assets/46312751/b40ec53d-914c-4735-9f94-26c60890d377">
